rootless.h

```diff
@@ -1,11 +1,11 @@
-#ifdef THEOS_PACKAGE_INSTALL_PREFIX
+#ifdef XINA_SUPPORT // Only define this for rootful compilations that need support for xina
+#define ROOT_PATH(cPath) [[NSFileManager defaultManager] fileExistsAtPath:@"/var/LIY"] ? "/var/jb" cPath : cPath
```

Comment:
This is a macro developers may expect to be able to use in a C file (i.e. file compiled in C language), however this macro requires Objective-C and the Foundation framework.

Reply:
Yes. But there is no other way to make this macro work otherwise. If you can think of any, I'm open to suggestions.

Reply:
Nvm I have an idea

Reply:
Lightmann helped with this idea:

```c
#define _THEOS_FILE_EXISTS(path) (faccessat(AT_FDCWD, path, F_OK, AT_SYMLINK_NOFOLLOW) == 0)
```

I'm still apprehensive about having a syscall in a macro that we expect to be commonly used, however if this is limited to e.g. Xina and it's needed, then I think we can move forward with this.

Reply:
That was not the problematic part. The problematic part was adding the prefix to the string without allocating any memory, I solved it in the new commit now by putting the output string on the stack.